### DISCUSSION TITLE
Allow TextColumn descriptions to be rendered as html

### DISCUSSION THREAD

**Discussion by dreftas** [filamentphp/filament#11725] [Ideas]
It would be nice to have possibility to allow TextColumn descriptions to be rendered as HTML.

**Comment by dissto** [comment]
I'm pretty sure you can just do that.

```php
TextColumn::make('name')
  ->description(fn () => new HtmlString('<span class="text-red-500">hi</span>'))
```

**Reply by dreftas** [reply]
No, we can't.
Blade view for text column does not parse HTML tags.  You can see in text-column.blade.php
[Here](https://github.com/filamentphp/filament/blob/3.x/packages/tables/resources/views/columns/text-column.blade.php#L74)
and
[Here](https://github.com/filamentphp/filament/blob/3.x/packages/tables/resources/views/columns/text-column.blade.php#L306)

There is no control whether output using `{!! !!}`, or `{{ }}` so it just shows HTML tags.

**Reply by dissto** [reply]
Mhh. Maybe i misunderstood but:

```php
 Tables\Columns\TextColumn::make('id')
      ->description(function () {
          return view('test');
      })
      ->label('ID')
      ->searchable(),
```

```php
<!-- test.blade.php -->
<div class="flex flex-col gap-x-2 prose w-full">
    <div>hi?
        <x-filament::icon icon="heroicon-o-user" class="w-4 h-4"/>
    </div>
    <div>
        <h1>hi</h1>
    </div>
    <div>
        <h2>hi</h2>
    </div>
</div>

```

Gives me:
![Screenshot 2024-03-07 134941](https://github.com/filamentphp/filament/assets/11778632/b6ae8ccf-d9e1-47d8-9af9-399c817216ff)

**Reply by dreftas** [reply]
Heh.. Everything ok, my mistake...
I've been using like this:
`->description(fn (MyModel $record) => $record->myHtmlString)`

and in model:
`public function getMyHtmlStringAttribute(): string
{
return new HtmlString('test<br>test');
}
`

And it printed out `test<br>test` in description, and this was all because my `getMyHtmlStringAttribute()` return type was declared as string, so I believe returned HtmlString() were converted back to string. 

Anyway, declaring return type as HtmlString worked. :)

**Comment by emre-tarhan** [comment]
hi. i had the same problem and solved it like this:
`->getStateUsing(function ($record) { return strip_tags($record->body); })`
